plugins/backstage-plugin-prometheus/src/hooks/usePrometheus.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useApi } from '@backstage/core-plugin-api';
+import { useAsync } from 'react-use';
+import { prometheusApiRef } from '../api';
+import {
+  PrometheusDisplayableAlert,
+  PrometheusMatrixVectorResult,
+  PrometheusResponse,
+  PrometheusRuleResponse,
+  PrometheusScalarStringResult,
+  ResultType,
+} from '../types';
+import _ from 'lodash';
+import { DateTime } from 'luxon';
+
+function isMetricResult(
+  resultType: ResultType,
+  _result: PrometheusMatrixVectorResult[] | PrometheusScalarStringResult[],
+): _result is PrometheusMatrixVectorResult[] {
+  return resultType === 'matrix' || resultType === 'vector';
+}
+
+function resultToGraphData(
+  result: PrometheusMatrixVectorResult[],
+  dimension?: string,
+) {
+  const keys: string[] = [];
+  const metrics: { [k: string]: any } = {};
+  if (result.length === 0) {
+    return { keys, metrics, data: [] };
+  }
+
+  const grouper =
+    dimension && result.every(it => Object.keys(it.metric).includes(dimension))
+      ? dimension
+      : Object.keys(result[0].metric)[0];
+
+  const data = _(result)
```

Comment:
GL someone who tries to fix anything on this chain. Godspeed

Reply:
Ouffft!